main.go

```diff
@@ -76,16 +79,27 @@ func main() {
 		os.Exit(1)
 	}
 
+	// If a watch namespace is detected (i.e. operator is namespace scoped), then pass the NS to cache.Options.DefaultNamespaces
+	// If no watch namespace is detected (i.e. operator is cluster scoped), then pass an empty Cache object
+	var managerCache = cache.Options{}
+	if namespace != "" {
+		managerCache = cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				namespace: {},
+			},
+		}
+	}
+
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Namespace:          namespace,
-		Scheme:             scheme,
-		MetricsBindAddress: metricsAddr,
-		Port:               9443,
-		LeaderElection:     enableLeaderElection,
-		LeaderElectionID:   "988b4062.3scale.net",
+		Cache:            managerCache,
+		Scheme:           scheme,
+		Metrics:          metricsserver.Options{BindAddress: metricsAddr},
+		WebhookServer:    webhook.NewServer(webhook.Options{Port: 9443}),
```

Comment:
It is unrelated, but do we use and feature for the webhook server? could we disable it?

Reply:
I'm provisioning a cluster now to confirm but I think you are correct and we can disable it

Reply:
I confirmed we can remove it and pushed the changes